Pixel local reco: reduce code duplication between SiPixelDigiErrorsFromSoA and SiPixelRawToDigi

czangela opened the pull request:
#### PR description:

This PR reduces code duplication between `SiPixelDigiErrorsFromSoA` and `SiPixelRawToDigi` (see #32483).

The FED errors are sorted into different collections for later usage, this part of the code was implemented by both of the classes above. Now it has been moved to `PixelDataFormatter` and is accessible through a member function `unpackFEDErrors(...)`.

Also there is some (harmless ?) cleanup done in `SiPixelRawToDigi` and `PixelDataFormatter`.

#### PR validation:

The code compiles and runs on the workflows `136.8855` and `136.885502` (phase 1) without any error.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

<!-- Please replace this text with any link to  -->

czangela committed:
shorten syntax for default error list vectors

czangela committed:
get rid of unnecessary debug variable

czangela committed:
remove cablingMapLabel from members

czangela committed:
remove unused label

czangela committed:
remove timer

I know it might be breaking, that's why this is an atomic commit and can be reversed easily -> git rebase

czangela committed:
Move unpacking of FED error collections to PixelDataFormatter

czangela committed:
take errors with const reference

czangela committed:
remove obsolete stuff from PixelDataFormatter

czangela committed:
use own typedefs

czangela committed:
apply naming convention for SiPixelRawToDigi members

czangela committed:
use initializer list and make members const

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22288

- This PR adds an extra 32KB to repository

cmsbuild commented:
A new Pull Request was created by  @czangela  for master.

It involves the following packages:

EventFilter/SiPixelRawToDigi


@perrotta, @jpata, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@OzAmram, @makortel, @VinInn, @Martin-Grunewald, @dkotlins, @ferencek, @tvami this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT AddOn 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/summary.html
**COMMIT**: 1379504d5d4e4926ccf03aa54d1e066373116fab
**CMSSW**: CMSSW_12_0_X_2021-04-26-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33526/14578/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrix-results/11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
```
----- Begin Fatal Exception 26-Apr-2021 15:56:30 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
Illegal parameter found in configuration.  The parameter is named:
 'Timing'
You could be trying to use a parameter name that is not
allowed for this plugin or it could be misspelled.
----- End Fatal Exception -------------------------------------------------
```
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrix-results/12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
```
----- Begin Fatal Exception 26-Apr-2021 15:57:18 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
Illegal parameter found in configuration.  The parameter is named:
 'Timing'
You could be trying to use a parameter name that is not
allowed for this plugin or it could be misspelled.
----- End Fatal Exception -------------------------------------------------
```
- [11634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrix-results/11634.911_TTbar_14TeV+2021_DD4hep+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
```
----- Begin Fatal Exception 26-Apr-2021 15:57:35 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
Illegal parameter found in configuration.  The parameter is named:
 'Timing'
You could be trying to use a parameter name that is not
allowed for this plugin or it could be misspelled.
----- End Fatal Exception -------------------------------------------------
```


## RelVals-INPUT

- [158.2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/158.2_PhotonJets_Pt_10_13_HI+PhotonJets_Pt_10_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO)```158.2_PhotonJets_Pt_10_13_HI+PhotonJets_Pt_10_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_PhotonJets_Pt_10_13_HI+PhotonJets_Pt_10_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO.log```
- [158.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/158.1_QCD_Pt_80_120_13_HI+QCD_Pt_80_120_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO)```158.1_QCD_Pt_80_120_13_HI+QCD_Pt_80_120_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_QCD_Pt_80_120_13_HI+QCD_Pt_80_120_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO.log```
- [158.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/158.3_ZEEMM_13_HI+ZEEMM_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO)```158.3_ZEEMM_13_HI+ZEEMM_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_ZEEMM_13_HI+ZEEMM_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [11603.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11603.0_SingleElectronPt1000+2021+SingleElectronPt1000_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11601.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11601.0_SingleElectronPt10+2021+SingleElectronPt10_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11602.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11602.0_SingleElectronPt35+2021+SingleElectronPt35_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11604.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11604.0_SingleGammaPt10+2021+SingleGammaPt10_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11605.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11605.0_SingleGammaPt35+2021+SingleGammaPt35_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11609.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11609.0_SingleMuPt1000+2021+SingleMuPt1000_Eta2p85_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11608.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11608.0_SingleMuPt100+2021+SingleMuPt100_Eta2p85_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11607.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11607.0_SingleMuPt10+2021+SingleMuPt10_Eta2p85_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11606.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11606.0_SingleMuPt1+2021+SingleMuPt1_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11630.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11630.0_QCD_Pt_3000_3500_14TeV+2021+QCD_Pt_3000_3500_14TeV_TuneCUETP8M1_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.1_TTbar_14TeV+2021_trackingOnly+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.24](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.24_TTbar_14TeV+2021_0T+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.5_TTbar_14TeV+2021_pixelTrackingOnly+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.505](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.505_TTbar_14TeV+2021_Patatrack_PixelOnlyTripletsCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.501](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.501_TTbar_14TeV+2021_Patatrack_PixelOnlyCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.511](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.511_TTbar_14TeV+2021_Patatrack_ECALOnlyCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.521](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.521_TTbar_14TeV+2021_Patatrack_HCALOnlyCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.7_TTbar_14TeV+2021_trackingMkFit+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.91](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11634.91_TTbar_14TeV+2021_trackdnn+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11646.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11646.0_ZEE_14+2021+ZEE_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11650.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11650.0_ZMM_14+2021+ZMM_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11640.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11640.0_MinBias_14TeV+2021+MinBias_14TeV_pythia8_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11643.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11643.0_QCD_FlatPt_15_3000HS_14+2021+QCDForPF_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11725.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11725.0_GluGluTo2Jets_14TeV+2021+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11650.505](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11650.505_ZMM_14+2021_Patatrack_PixelOnlyTripletsCPU+ZMM_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11723.17](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11723.17_QCD_Pt_1800_2400_14+2021_seedingDeepCore+QCD_Pt_1800_2400_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11650.501](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11650.501_ZMM_14+2021_Patatrack_PixelOnlyCPU+ZMM_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11834.0_TTbar_14TeV+2021PU+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11834.24](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11834.24_TTbar_14TeV+2021PU_0T+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11846.0_ZEE_14+2021PU+ZEE_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11834.13](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11834.13_TTbar_14TeV+2021PU_mlpf+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [12034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/12034.0_TTbar_14TeV+2021Design+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11925.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/11925.0_GluGluTo2Jets_14TeV+2021PU+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [12634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/12634.0_TTbar_14TeV+2023PU+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/12834.0_TTbar_14TeV+2024+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/runTheMatrixINPUT-results/13034.0_TTbar_14TeV+2024PU+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
</details>



## AddOn Tests

- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_mc_PRef)
```
----- Begin Fatal Exception 26-Apr-2021 15:41:32 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
Illegal parameter found in configuration.  The parameter is named:
 'Timing'
You could be trying to use a parameter name that is not
allowed for this plugin or it could be misspelled.
----- End Fatal Exception -------------------------------------------------
```
- [hlt_data_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_data_PRef)
```
----- Begin Fatal Exception 26-Apr-2021 15:41:19 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
Illegal parameter found in configuration.  The parameter is named:
 'Timing'
You could be trying to use a parameter name that is not
allowed for this plugin or it could be misspelled.
----- End Fatal Exception -------------------------------------------------
```
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_mc_HIon)
```
----- Begin Fatal Exception 26-Apr-2021 15:42:12 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
Illegal parameter found in configuration.  The parameter is named:
 'Timing'
You could be trying to use a parameter name that is not
allowed for this plugin or it could be misspelled.
----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more addon errors ...</summary>

- [hlt_data_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_data_HIon)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_mc_GRun)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_mc_GRun)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_mc_PRef)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_mc_PRef)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_mc_HIon)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_mc_HIon)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_mc_GRun)
- [hlt_data_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14578/addOnTests/hlt_data_GRun)
</details>

czangela committed:
Use EDPutTokens in SiPixelRawData
- also, add Timing as an optional
- manage header includes to minimize dependency

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22291

- This PR adds an extra 16KB to repository

cmsbuild commented:
Pull request #33526 was updated. @perrotta, @jpata, @cmsbuild, @slava77 can you please check and sign again.

slava77 commented:
enable gpu

slava77 commented:
@cmsbuild please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT AddOn 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/summary.html
**COMMIT**: ac54098e42d664dc72882872536d57a61f4fee36
**CMSSW**: CMSSW_12_0_X_2021-04-26-1100/slc7_amd64_gcc900
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33526/14583/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrix-results/11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
```
----- Begin Fatal Exception 26-Apr-2021 17:53:22 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
In the configuration, parameter "Timing" is defined as an untracked bool.
It should be tracked.
----- End Fatal Exception -------------------------------------------------
```
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrix-results/12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
```
----- Begin Fatal Exception 26-Apr-2021 17:53:55 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
In the configuration, parameter "Timing" is defined as an untracked bool.
It should be tracked.
----- End Fatal Exception -------------------------------------------------
```
- [11634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrix-results/11634.911_TTbar_14TeV+2021_DD4hep+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
```
----- Begin Fatal Exception 26-Apr-2021 17:54:11 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
In the configuration, parameter "Timing" is defined as an untracked bool.
It should be tracked.
----- End Fatal Exception -------------------------------------------------
```


## RelVals-INPUT

- [158.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/158.1_QCD_Pt_80_120_13_HI+QCD_Pt_80_120_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO)```158.1_QCD_Pt_80_120_13_HI+QCD_Pt_80_120_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_QCD_Pt_80_120_13_HI+QCD_Pt_80_120_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO.log```
- [158.2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/158.2_PhotonJets_Pt_10_13_HI+PhotonJets_Pt_10_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO)```158.2_PhotonJets_Pt_10_13_HI+PhotonJets_Pt_10_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_PhotonJets_Pt_10_13_HI+PhotonJets_Pt_10_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO.log```
- [158.3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/158.3_ZEEMM_13_HI+ZEEMM_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO)```158.3_ZEEMM_13_HI+ZEEMM_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO/step2_ZEEMM_13_HI+ZEEMM_13_HIINPUT+DIGIHI2018PPRECO+RECOHI2018PPRECO+HARVESTHI2018PPRECO.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [11601.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11601.0_SingleElectronPt10+2021+SingleElectronPt10_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11602.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11602.0_SingleElectronPt35+2021+SingleElectronPt35_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11603.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11603.0_SingleElectronPt1000+2021+SingleElectronPt1000_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11604.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11604.0_SingleGammaPt10+2021+SingleGammaPt10_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11605.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11605.0_SingleGammaPt35+2021+SingleGammaPt35_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11609.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11609.0_SingleMuPt1000+2021+SingleMuPt1000_Eta2p85_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11608.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11608.0_SingleMuPt100+2021+SingleMuPt100_Eta2p85_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11606.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11606.0_SingleMuPt1+2021+SingleMuPt1_pythia8_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11607.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11607.0_SingleMuPt10+2021+SingleMuPt10_Eta2p85_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.24](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.24_TTbar_14TeV+2021_0T+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.505](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.505_TTbar_14TeV+2021_Patatrack_PixelOnlyTripletsCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.1_TTbar_14TeV+2021_trackingOnly+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11630.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11630.0_QCD_Pt_3000_3500_14TeV+2021+QCD_Pt_3000_3500_14TeV_TuneCUETP8M1_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11634.501](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.501_TTbar_14TeV+2021_Patatrack_PixelOnlyCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.5_TTbar_14TeV+2021_pixelTrackingOnly+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.511](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.511_TTbar_14TeV+2021_Patatrack_ECALOnlyCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.7_TTbar_14TeV+2021_trackingMkFit+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.521](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.521_TTbar_14TeV+2021_Patatrack_HCALOnlyCPU+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11634.91](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11634.91_TTbar_14TeV+2021_trackdnn+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11646.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11646.0_ZEE_14+2021+ZEE_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11640.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11640.0_MinBias_14TeV+2021+MinBias_14TeV_pythia8_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11650.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11650.0_ZMM_14+2021+ZMM_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11643.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11643.0_QCD_FlatPt_15_3000HS_14+2021+QCDForPF_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11723.17](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11723.17_QCD_Pt_1800_2400_14+2021_seedingDeepCore+QCD_Pt_1800_2400_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11834.24](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11834.24_TTbar_14TeV+2021PU_0T+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11834.13](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11834.13_TTbar_14TeV+2021PU_mlpf+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11834.0_TTbar_14TeV+2021PU+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11725.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11725.0_GluGluTo2Jets_14TeV+2021+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [11650.501](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11650.501_ZMM_14+2021_Patatrack_PixelOnlyCPU+ZMM_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11650.505](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11650.505_ZMM_14+2021_Patatrack_PixelOnlyTripletsCPU+ZMM_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [11846.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11846.0_ZEE_14+2021PU+ZEE_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [11925.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/11925.0_GluGluTo2Jets_14TeV+2021PU+GluGluTo2Jets_M_300_2000_14TeV_Exhume_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [12034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/12034.0_TTbar_14TeV+2021Design+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [12634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/12634.0_TTbar_14TeV+2023PU+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
- [12834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/12834.0_TTbar_14TeV+2024+TTbar_14TeV_TuneCP5_GenSimINPUT+Digi+Reco+HARVEST+ALCA)
- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/runTheMatrixINPUT-results/13034.0_TTbar_14TeV+2024PU+TTbar_14TeV_TuneCP5_GenSimINPUT+DigiPU+RecoPU+HARVESTPU+Nano)
</details>



## AddOn Tests

- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_mc_PRef)
```
----- Begin Fatal Exception 26-Apr-2021 17:40:37 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
In the configuration, parameter "Timing" is defined as an untracked bool.
It should be tracked.
----- End Fatal Exception -------------------------------------------------
```
- [hlt_data_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_data_PRef)
```
----- Begin Fatal Exception 26-Apr-2021 17:37:23 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
In the configuration, parameter "Timing" is defined as an untracked bool.
It should be tracked.
----- End Fatal Exception -------------------------------------------------
```
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_mc_HIon)
```
----- Begin Fatal Exception 26-Apr-2021 17:40:42 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiPixelRawToDigi label='hltSiPixelDigis'
Exception Message:
In the configuration, parameter "Timing" is defined as an untracked bool.
It should be tracked.
----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more addon errors ...</summary>

- [hlt_data_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_data_HIon)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_mc_GRun)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_mc_GRun)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_mc_PRef)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_mc_PRef)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_mc_HIon)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_mc_HIon)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_mc_GRun)
- [hlt_data_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14583/addOnTests/hlt_data_GRun)
</details>



## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 4
 - DQMHistoTests: Total histograms compared: 9559
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 9559
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 3 files compared)
 - Checked 12 log files, 9 edm output root files, 4 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_12_0_X_2021-04-26-1100+fa1837/42410/triggerResults): no differences found

Martin-Grunewald commented:
Add some temporary HLT customisation here: https://github.com/cms-sw/cmssw/blob/master/HLTrigger/Configuration/python/customizeHLTforCMSSW.py, deleting this parameter for all instances of the plugin (I'll remove the customisation once the HLT dumps as extracted from ConfDB have been updated).

czangela committed:
remove Timing from HLT menus

extra validation: ran addOnTests.py and unittests locally

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22310

- This PR adds an extra 68KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #33495

cmsbuild commented:
Pull request #33526 was updated. @perrotta, @cmsbuild, @slava77, @Martin-Grunewald, @fwyzard, @jpata can you please check and sign again.

mmusich commented on HLTrigger/Configuration/python/customizeHLTforCMSSW.py:
```diff
@@ -129,11 +129,20 @@ def customiseFor2018Input(process):
 
     return process
 
+def customizeForPixelRawToDigi(process):
```
please call it `customiseFor33526`

mmusich commented on EventFilter/SiPixelRawToDigi/plugins/SiPixelRawToDigi.cc:
```diff
@@ -103,9 +103,7 @@ void SiPixelRawToDigi::fillDescriptions(edm::ConfigurationDescriptions& descript
-  // unused options, kept for back-compatibility
-  desc.addOptional<bool>("CheckPixelOrder");
-  desc.addOptional<bool>("Timing");
+  desc.addOptional<bool>("CheckPixelOrder");  // unused option, kept for back-compatibility
```
perhaps time to remove?

czangela committed:
remove unused variable CheckPixelOrder

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22311

- This PR adds an extra 72KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #33495

cmsbuild commented:
Pull request #33526 was updated. @perrotta, @cmsbuild, @slava77, @Martin-Grunewald, @fwyzard, @jpata can you please check and sign again.

mmusich commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14613/summary.html
**COMMIT**: 213287cbbd427158b17e1b37fa0418db8bf45a2d
**CMSSW**: CMSSW_12_0_X_2021-04-26-2300/slc7_amd64_gcc900
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33526/14613/install.sh` to create a dev area with all the needed externals and cmssw changes.


## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 4
 - DQMHistoTests: Total histograms compared: 9559
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 9559
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 3 files compared)
 - Checked 12 log files, 9 edm output root files, 4 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_12_0_X_2021-04-26-2300+fa1837/42430/triggerResults): no differences found


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 3036 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 2877605
 - DQMHistoTests: Total failures: 7702
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2869880
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 37 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 160 log files, 37 edm output root files, 38 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-04-26-2300+fa1837/42432/triggerResults): found differences in 1 / 37 workflows

slava77 commented:
> Reco comparison results: 3036 differences found in the comparisons

these are in the DD4HEP workflow 11634.911; something that we've seen before as random.
@cms-sw/geometry-l2 is there still some known non-repeatability in the DD4HEP workflows?

slava77 commented:
@cmsbuild please test
to see if the issue in 11634.911 goes away or sticks with this PR.

cvuosalo commented:
@slava77 We had thought we fixed the instability in DD4hep simulation results, but the test in this PR seems to indicate it may be occurring at a low frequency that makes it difficult to reproduce.
In any case, the comparison differences are merely statistical fluctuations and are not significant.

makortel commented:
> In any case, the comparison differences are merely statistical fluctuations and are not significant.

With (very) limited number of events it is difficult to distinguish statistical fluctuations from systematic changes. This is acceptable for Dd4hep workflows for now that they are not in production. But when they become, the current result compatibility testing becomes much less useful if there are occasional statistical fluctuations in many places.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14622/summary.html
**COMMIT**: 213287cbbd427158b17e1b37fa0418db8bf45a2d
**CMSSW**: CMSSW_12_0_X_2021-04-27-1100/slc7_amd64_gcc900
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33526/14622/install.sh` to create a dev area with all the needed externals and cmssw changes.


## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 4
 - DQMHistoTests: Total histograms compared: 9559
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 9559
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 3 files compared)
 - Checked 12 log files, 9 edm output root files, 4 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_12_0_X_2021-04-27-1100+fa1837/42443/triggerResults): no differences found


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 3034 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 2877605
 - DQMHistoTests: Total failures: 7706
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2869877
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 37 files compared)
 - Checked 160 log files, 37 edm output root files, 38 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-04-27-1100+fa1837/42439/triggerResults): found differences in 1 / 37 workflows

makortel commented:
> @cms-sw/geometry-l2 is there still some known non-repeatability in the DD4HEP workflows?

I opened a new issue #33552 to track this non-repeatibility.

czangela committed:
use range-based for loop for map iteration, increase readibility

best c++17 feature

czangela committed:
make put token const

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22340

- This PR adds an extra 16KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #33495, #33543

cmsbuild commented:
Pull request #33526 was updated. @perrotta, @cmsbuild, @slava77, @Martin-Grunewald, @fwyzard, @jpata can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22342

- This PR adds an extra 16KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #33495, #33543

mmusich commented:
allow @czangela  test rights

czangela commented:
@cmsbuild please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22350

- This PR adds an extra 80KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #33543

cmsbuild commented:
Pull request #33526 was updated. @perrotta, @cmsbuild, @slava77, @Martin-Grunewald, @fwyzard, @jpata can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22352

- This PR adds an extra 76KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #33543

cmsbuild commented:
Pull request #33526 was updated. @perrotta, @cmsbuild, @slava77, @Martin-Grunewald, @fwyzard, @jpata can you please check and sign again.

czangela commented:
@cmsbuild please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14678/summary.html
**COMMIT**: 2dc18482ac442e8874bff5cda47aefe9574161d0
**CMSSW**: CMSSW_12_0_X_2021-04-28-1100/slc7_amd64_gcc900
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33526/14678/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - cms-sw/cmssw#33437 @barvic: re-introduce IORawData/CSCCommissioning module for CSC data files conversion

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14678/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14678/git-merge-result


## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 4
 - DQMHistoTests: Total histograms compared: 9559
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 9559
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 3 files compared)
 - Checked 12 log files, 9 edm output root files, 4 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_12_0_X_2021-04-28-1100+fa1837/42478/triggerResults): no differences found


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 38
 - DQMHistoTests: Total histograms compared: 2877605
 - DQMHistoTests: Total failures: 9
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2877573
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 37 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 160 log files, 37 edm output root files, 38 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-04-28-1100+fa1837/42481/triggerResults): no differences found

slava77 commented:
@mmusich 
please clarify if this now looks good from the DPG point of view

mmusich commented on EventFilter/SiPixelRawToDigi/src/PixelDataFormatter.cc:
```diff
@@ -520,3 +406,76 @@ std::string PixelDataFormatter::print(const Word64& word) const {
+      for (auto const& aPixelError : errorDetSet) {
+        // For the time being, we extend the error handling functionality with ErrorType 25
+        // In the future, we should sort out how the usage of tkerrorlist can be generalized
+        if (aPixelError.getType() == 25) {
```
the meaning of FED Error 25 is different between phase-0 and phase-1, so the lines below should apply only for the phase-1 reconstruction.

mmusich commented:
@slava77 
@tsusa and I are reviewing, will let you know once its ready to go.

slava77 commented on EventFilter/SiPixelRawToDigi/plugins/SiPixelRawToDigi.h:
```diff
@@ -10,15 +10,23 @@
 #include "FWCore/Framework/interface/stream/EDProducer.h"
```
we've been asking since a while to not make header files for (concrete) plugins. Since this file is rather heavily modified already, please conform to having no header as well and merge the content here in SiPixelRawToDigi.cc

czangela committed:
fix phase 1 condition

czangela committed:
remove header for plugin SiPixelRawToDigi

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22388

- This PR adds an extra 72KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #33543

cmsbuild commented:
Pull request #33526 was updated. @perrotta, @cmsbuild, @slava77, @Martin-Grunewald, @fwyzard, @jpata can you please check and sign again.

czangela committed:
remove unnecessary header includes, forward declarations

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33526/22390

- This PR adds an extra 72KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #33543

cmsbuild commented:
Pull request #33526 was updated. @perrotta, @cmsbuild, @slava77, @Martin-Grunewald, @fwyzard, @jpata can you please check and sign again.

fwyzard commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14735/summary.html
**COMMIT**: 2e7301b193cdc96c354b65b061e7b8042d925b81
**CMSSW**: CMSSW_12_0_X_2021-04-29-2300/slc7_amd64_gcc900
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33526/14735/install.sh` to create a dev area with all the needed externals and cmssw changes.


## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 4
 - DQMHistoTests: Total histograms compared: 9559
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 9559
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 3 files compared)
 - Checked 12 log files, 9 edm output root files, 4 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_12_0_X_2021-04-29-2300+fa1837/42515/triggerResults): no differences found


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2662646
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2662623
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 36 files compared)
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-04-29-2300+fa1837/42518/triggerResults): no differences found

cmsbuild commented:
Pull request #33526 was updated. @perrotta, @cmsbuild, @slava77, @Martin-Grunewald, @fwyzard, @jpata can you please check and sign again.

czangela commented:
@cmsbuild  please test
-- had to rebase due to a conflict

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14747/summary.html
**COMMIT**: 64d828b400c67c5145190f855b75579c99654aac
**CMSSW**: CMSSW_12_0_X_2021-04-29-2300/slc7_amd64_gcc900
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33526/14747/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - cms-sw/cmssw#33560 @davidlange6: remove Python2 unit tests ; migrate scripts using uproot to use python3
 - cms-sw/cmssw#33543 @jeongeun: Migrate module configurations in RecoLocalMuon to use automatically generated cfi default configurations
 - cms-sw/cmssw#31697 @makortel: Migrate most of MixingModule and PreMixingModule to EventSetup consumes
 - cms-sw/cmssw#33520 @hatakeyamak: Adopt HFSignalAsymmetry flag in HF PFRecHit cleaning
 - cms-sw/cmssw#33386 @vjmastra: EtaBToJpsiJpsi gen fragment at 14 TeV for release validation

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14747/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14747/git-merge-result


## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 4
 - DQMHistoTests: Total histograms compared: 9559
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 9559
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 3 files compared)
 - Checked 12 log files, 9 edm output root files, 4 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_12_0_X_2021-04-29-2300+fa1837/42526/triggerResults): no differences found


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 1014 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2662646
 - DQMHistoTests: Total failures: 1155
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2661468
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.016 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 136.874 ): -0.016 KiB       JetMET/SUSYDQM
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-04-29-2300+fa1837/42532/triggerResults): no differences found

slava77 commented:
@cmsbuild please test

to get a cleaner set (even though 30-2300 is not out yet; advancing to 30-1100 should be enough to get rid of differences in reco)

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14770/summary.html
**COMMIT**: 64d828b400c67c5145190f855b75579c99654aac
**CMSSW**: CMSSW_12_0_X_2021-04-30-1100/slc7_amd64_gcc900
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33526/14770/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - cms-sw/cmssw#33543 @jeongeun: Migrate module configurations in RecoLocalMuon to use automatically generated cfi default configurations

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14770/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-fa1837/14770/git-merge-result


## GPU Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 4
 - DQMHistoTests: Total histograms compared: 9559
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 9559
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 3 files compared)
 - Checked 12 log files, 9 edm output root files, 4 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsGPU/CMSSW_12_0_X_2021-04-30-1100+fa1837/42553/triggerResults): no differences found


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2662646
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2662623
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 36 files compared)
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-04-30-1100+fa1837/42555/triggerResults): no differences found

mmusich commented:
It should be now good to go from the DPG perspective.

slava77 commented:
+reconstruction

for https://github.com/cms-sw/cmssw/pull/33526	64d828b 
- code changes are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no differences

Martin-Grunewald commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

qliphy commented:
+1

makortel commented:
Seems that this PR caused a test failure in the IBs, see https://github.com/cms-sw/cmssw/issues/33623

mmusich commented:
Issue is addressed at https://github.com/cms-sw/cmssw/pull/33625

@czangela there some other occurrences of this parameters in other (untested) files: 

https://cmssdt.cern.ch/dxr/CMSSW/search?q=siPixelDigis.Timing&case=true&redirect=true

perhaps you can clean those up to in a follow-up PR.
